maven/lib/dependabot/maven/new_version.rb

```diff
@@ -0,0 +1,68 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "dependabot/maven/version_parser"
+require "dependabot/version"
+require "dependabot/utils"
+
+# See https://maven.apache.org/pom.html#Version_Order_Specification for details.
+
+module Dependabot
+  module Maven
+    class NewVersion
+      extend T::Sig
+      extend T::Helpers
+
+      PRERELEASE_QUALIFIERS = T.let([
+        Dependabot::Maven::VersionParser::ALPHA,
+        Dependabot::Maven::VersionParser::BETA,
+        Dependabot::Maven::VersionParser::MILESTONE,
+        Dependabot::Maven::VersionParser::RC,
+        Dependabot::Maven::VersionParser::SNAPSHOT
+      ].freeze, T::Array[Integer])
+
+      sig { returns(Dependabot::Maven::TokenBucket) }
+      attr_accessor :token_bucket
+
+      sig { params(version: T.nilable(String)).returns(T::Boolean) }
```

Comment:
any reason to have nilable String in here? Can we rely on strong Sorbet typing to ensure callers always pass in string values?

Reply:
But we've had many cases where nil versions are created from update_checker, version finder etc. Unless we fix it from there, I think I could end up with runtime errors when those pass nil as the param.

This way at least I'm handling that scenario explicitly. If I remove nilable then I'll have to remove the guard clause too.